### PR TITLE
Make torch.equal work with boolean CPU tensors

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: aten/src/ATen/Declarations.cwrap [hunk 1]
```diff
@@ -316,6 +316,8 @@
 ]]
 [[
   name: _th_equal
+  cpu_bool: True
+  cuda_bool: True
```
Comment (izdeby) [inline aten/src/ATen/Declarations.cwrap]: You will have to either remove cuda_bool or add cuda related changes, which is more preferable.

**Review by izdeby — CHANGES_REQUESTED** [review]

File: aten/src/ATen/Declarations.cwrap (hunk 1 above)
Comment (asuhan) [reply aten/src/ATen/Declarations.cwrap]: Done, CUDA is a bit harder to test for me.

**Comment by izdeby** [comment]
Please, make sure that all tests are passing and should be good to go

**Review by izdeby — APPROVED** [review]

**Comment by asuhan** [comment]
Thanks! Build failure looks unrelated.

**Review by li-roy — APPROVED** [review]

**Pull request closed without merging**